rusoto/core/Cargo.toml

```diff
@@ -26,7 +26,7 @@ rustc_version = "0.2.1"
 [dependencies]
 futures = "0.1.16"
 hmac = "0.5.0"
-hyper = "0.12.0"
+hyper = "0.12.16"
```

Comment:
I believe the existing `0.12.0` specification will pick up the latest version, `0.12.16`, when it's being built. Does this change help that by not requiring Rusoto users to delete their Cargo.lock file or run `cargo update`?

Reply:
Does `0.12.0` do it, or would it need to be `0.12`?

I interpreted `0.12.0` as specifying exactly `0.12.0` which is why I specified `0.12.16`

Reply:
This often trips me up as well. 😄  The [latest cargo guide on specifying dependencies](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html) has improved in clarity:

```
[dependencies]
time = "0.1.12"
```

> The string "0.1.12" is a semver version requirement. Since this string does not have any operators in it, it is interpreted the same way as if we had specified "^0.1.12", which is called a caret requirement.
...
> Caret requirements allow SemVer compatible updates to a specified version. An update is allowed if the new version number does not modify the left-most non-zero digit in the major, minor, patch grouping. In this case, if we ran cargo update -p time, cargo should update us to version 0.1.13 if it is the latest 0.1.z release, but would not update us to 0.2.0.

If we really wanted hyper 0.12.0 and only 0.12.0, we could specify it like this:

```
hyper = "=0.12.16"
```

Hope this makes sense! 😄

Reply:
I changed it to 
```
hyper = "0.12"
```
to be more clear that we're defaulting to the latest patch version of 0.12